Client/core/Graphics/CGraphics.cpp

```diff
@@ -483,50 +485,59 @@ void CGraphics::EndDrawBatch()
 //
 void CGraphics::CheckModes(EDrawModeType newDrawMode, EBlendModeType newBlendMode)
 {
-    bool bDrawModeChanging = (m_CurDrawMode != newDrawMode);
-    bool bBlendModeChanging = (m_CurBlendMode != newBlendMode);
-    // Draw mode changing?
-    if (bDrawModeChanging || bBlendModeChanging)
     {
+        const bool bDrawModeChanging = (m_CurDrawMode != newDrawMode);
+        const bool bBlendModeChanging = (m_CurBlendMode != newBlendMode);
 
-        // Flush old
-        if (m_CurDrawMode == EDrawMode::DX_SPRITE)
-        {
-            m_pDevice->SetSamplerState(0, D3DSAMP_ADDRESSU, D3DTADDRESS_WRAP);
-            m_pDevice->SetSamplerState(0, D3DSAMP_ADDRESSV, D3DTADDRESS_WRAP);
-            m_pDXSprite->End();
-        }
-        else if (m_CurDrawMode == EDrawMode::DX_LINE)
-        {
-            m_pLineInterface->End();
-        }
-        else if (m_CurDrawMode == EDrawMode::TILE_BATCHER)
-        {
-            m_pTileBatcher->Flush();
-        }
-        else if (m_CurDrawMode == EDrawMode::PRIMITIVE)
-        {
-            m_pPrimitiveBatcher->Flush();
-        }
-        else if (m_CurDrawMode == EDrawMode::PRIMITIVE_MATERIAL)
-        {
-            m_pPrimitiveMaterialBatcher->Flush();
-        }
+        // Draw mode changing?
+        if (!bDrawModeChanging && !bBlendModeChanging)
+            return;
+    }
```

Comment:
What is the extra scope here for? Also there's not really a need for const here

Reply:
well, just wanted to indicate neither of the 2 bools are used later. Putting the in a scope indicates that.

Reply:
Resolved?